### ISSUE TITLE
Monai examples with Keras using torch as backend

### ISSUE THREAD

**Issue by emi-dm** [keras-team/keras#19090]
Does someone know if a Monai integration with Keras using a torch as a backend exists? I found this example: [https://colab.research.google.com/drive/1x8VjJXpkK7aYVrzt2I3eaYs9ewpHDaJr?usp=sharing](https://colab.research.google.com/drive/1x8VjJXpkK7aYVrzt2I3eaYs9ewpHDaJr?usp=sharing), but I'm interested in learning more about it.

Thanks in advance!

**Comment by SuryanarayanaY** [comment]
Hi @emi-research-dl ,

If the attached examples works with `keras_core` it should work with `keras3` also. I doubt only on complete serialization of the model due to external libraries.Otherwise, AFAIK we can train the model and can save and load weights using model.save_weights and model.load_weights respectively.

Could you please try and let us know the outcome !

**Comment by emi-research-dl** [comment]
Hi @SuryanarayanaY   serialization may not be a problem using a Keras model instead of a Monai model. The main problem could be the serialization of custom-used loss functions and metrics. Is there any way to serialize, for example, a torch loss function without saving the model weights and later loading them again? I think it would be an excellent feature for Keras, a serializer for custom objects.

**Comment by SuryanarayanaY** [comment]
Hi @emi-dm ,

If the custom object is of keras object there are ways to serialize and deserialize those which are briefed [here](https://keras.io/guides/serialization_and_saving/#model-serialization).

I have tested a custom model using model subclassing, with torch backend and APIs and trained it. After training, I have saved and reloaded the model and model.predict seems working fine as per [gist](https://colab.research.google.com/gist/SuryanarayanaY/bf7f59b3e3e48b580b8b6054d381f4fe/serialization_torch_model_custom_train_step.ipynb). Similarly you can also try to write a custom_loss function by overriding `compute_loss` method and let us know the outcome.

Thanks!

**Comment by emi-dm** [comment]
Thanks @SuryanarayanaY , but I was reffering for the case in which you use a torch loss function!

**Comment by SuryanarayanaY** [comment]
Hi @emi-dm ,

You can use torch loss function such as `torch.nn.CrossEntropyLoss()` and with the help of `custom training loop` you can train the model.The model can be saved into `.keras` format and can reloaded again as it is.

 I have tested with torch loss function with custom training loop and then saved and reloaded the model and predictions are same for both.Reference [gist](https://colab.research.google.com/gist/SuryanarayanaY/05b0738f00bc5ec177ae5e15491b2280/19090_writing_a_custom_training_loop_in_torch.ipynb#scrollTo=anNXalZeKS_7). 

SImilar issue has been raised by you in #19084, can you please close this and track there?

Thanks!

**Comment by emi-dm** [comment]
Thank you so much @SuryanarayanaY!!!
